pkg/adaptor/xds/v3/route.go

```diff
@@ -87,6 +87,7 @@ func (adaptor *adaptor) translateVirtualHost(prefix string, vhost *routev3.Virtu
 		}
 		vars = append(vars, queryVars...)
 		name = fmt.Sprintf("%s#%s#%s", name, vhost.GetName(), prefix)
+		name = strings.Replace(name, ".svc.cluster.local", "", -1) // avoid name too long
```

Comment:
Just use `strings.TrimSuffix`.

Also, this works most of time but will out of work if the cluster domain is not `cluster.local`.

Reply:
`.svc.cluster.local` is not configurable now
https://github.com/api7/apisix-mesh-agent/blob/309cdb5a16dc06b15965de47939442aacf27fb51/pkg/provisioner/xds/v3/grpc/types.go#L86-L87

Reply:
`strings.TrimSuffix` is not enough. vhost.Name could be FQDN.

Reply:
In which case `.svc.cluster.local` would be inside the FQDN but not as the suffix?

Reply:
Name is concated with schema `"%s#%s#%s"`

Reply:
OK, I forgot this detail...